judge/models/tests/test_contest.py

```diff
@@ -0,0 +1,454 @@
+        self.users.update({
+            'staff_contest_edit_own': create_user(
+                username='staff_contest_edit_own',
+                is_staff=True,
+                user_permissions=('edit_own_contest',),
+            ),
+            'staff_contest_see_all': create_user(
+                username='staff_contest_see_all',
+                user_permissions=('see_private_contest',),
+            ),
+            'staff_contest_edit_all': create_user(
+                username='staff_contest_edit_all',
+                is_staff=True,
+                user_permissions=('edit_own_contest', 'edit_all_contest'),
+            ),
+        })
+
+        _now = timezone.now()
+
+        self.basic_contest = create_contest(
+            key='basic',
+            start_time=_now - timezone.timedelta(days=1),
+            end_time=_now + timezone.timedelta(days=100),
+            organizers=('superuser', 'staff_contest_edit_own'),
+        )
+
+        self.hidden_scoreboard_contest = create_contest(
+            key='hidden_scoreboard',
+            start_time=_now - timezone.timedelta(days=1),
+            end_time=_now + timezone.timedelta(days=100),
+            is_visible=True,
+            hide_scoreboard=True,
+            problem_label_script='''
+                function(n)
+                    return tostring(math.floor(n))
+                end
+            ''',
+        )
+
+        self.profiles['normal'].current_contest = create_contest_participation(
+            contest='hidden_scoreboard',
+            user='normal',
+        )
+        self.profiles['normal'].save()
+
+        self.hidden_scoreboard_contest.update_user_count()
+
+        self.private_contest = create_contest(
+            key='private',
+            start_time=_now - timezone.timedelta(days=5),
+            end_time=_now - timezone.timedelta(days=3),
+            is_visible=True,
+            is_private=True,
+            is_organization_private=True,
+            private_contestants=('staff_contest_edit_own',),
+        )
+
+        self.organization_private_contest = create_contest(
+            key='organization_private',
+            start_time=_now + timezone.timedelta(days=3),
+            end_time=_now + timezone.timedelta(days=6),
+            is_visible=True,
+            is_organization_private=True,
+            organizations=('open',),
+            view_contest_scoreboard=('normal',),
+        )
+
+        self.private_user_contest = create_contest(
+            key='private_user',
+            start_time=_now - timezone.timedelta(days=3),
+            end_time=_now + timezone.timedelta(days=6),
+            is_visible=True,
+            is_private=True,
+        )
+
+    def setUp(self):
+        self.profiles['normal'].refresh_from_db()
+
+    def test_basic_contest(self):
+        self.assertTrue(self.basic_contest.show_scoreboard)
+        self.assertIsInstance(self.basic_contest.contest_window_length, timezone.timedelta)
```

Comment:
Why not assert the actual values?

Reply:
Done for this (since the `timedelta` is fixed). However, I don't think we should assert the actual values for the ones that are dependant on the current time (e.g `time_remaining)`.

Reply:
Could you not get an accurate calculation by using the Contest's `_now` cached property?

Reply:
Personally, I don't think it is worth it to check for the exact match in these cases. If you're using the Contest's `_now` property, you basically end up rewriting that method's calculation as a test, which doesn't really test what the method "should" return.